llvm/lib/Transforms/IPO/ElimAvailExtern.cpp

```diff
@@ -30,13 +31,24 @@ using namespace llvm;
 STATISTIC(NumFunctions, "Number of functions removed");
 STATISTIC(NumVariables, "Number of global variables removed");
 
+static bool skipGlobalObject(const GlobalObject &GO) {
+  if (const auto *const T = GO.getMetadata(LLVMContext::MD_repo_ticket)) {
+    if (const TicketNode *const MD = dyn_cast<TicketNode>(T)) {
```

Comment:
Is there a reason why you use auto for the declaration above (where the type isn't obvious without looking at the definition of getMetadata()) but provide an explicit type for the result of dyn_cast<> (where the type is the template argument)?

Reply:
Good point. Changed `auto` to `MDNode`.